components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1015,249 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
+    }
+
+    /**
+     * Validate sortBy.
+     *
+     * @param sortBy sortBy attribute.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortBy(String sortBy) {
+
+        if (StringUtils.isBlank(sortBy)) {
+            if (log.isDebugEnabled()) {
+                log.debug("sortBy attribute is empty. Therefore we set the default sortBy attribute. sortBy" +
+                        IdPManagementConstants.DEFAULT_SORT_BY);
+            }
+            return IdPManagementConstants.DEFAULT_SORT_BY;
+        } else {
+            switch (sortBy) {
+                case IdPManagementConstants.IDP_NAME:
+                    sortBy = IdPManagementConstants.NAME;
+                    break;
+                case IdPManagementConstants.IDP_HOME_REALM_ID:
+                    sortBy = IdPManagementConstants.HOME_REALM_ID;
+                    break;
+                default:
+                    sortBy = IdPManagementConstants.DEFAULT_SORT_BY;
+                    if (log.isDebugEnabled()) {
+                        log.debug("sortBy attribute is incorrect. Therefore we set the default sortBy attribute. " +
+                                "sortBy: " + IdPManagementConstants.DEFAULT_SORT_BY);
+                    }
+                    break;
+            }
+            return sortBy;
+        }
+    }
+
+    /**
+     * Validate sortOrder.
+     *
+     * @param sortOrder sortOrder ASC/DESC.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortOrder(String sortOrder) {
+
+        if (StringUtils.isBlank(sortOrder)) {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is empty. Therefore we set the default sortOrder value as ASC. SortOrder: " +
+                        sortOrder);
+            }
+        } else if (sortOrder.equals(IdPManagementConstants.DESC_SORT_ORDER)) {
+            sortOrder = IdPManagementConstants.DESC_SORT_ORDER;
+        } else if (sortOrder.equals(IdPManagementConstants.ASC_SORT_ORDER)) {
+            sortOrder = IdPManagementConstants.ASC_SORT_ORDER;
+        } else {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is incorrect. Therefore we set the default sortOrder value as ASC. SortOrder: "
+                        + sortOrder);
+            }
+        }
+        return sortOrder;
+    }
+
+    /**
+     * Validate limit.
+     *
+     * @param limit given limit value.
+     * @return validated limit and offset value.
+     */
+    private int validateLimit(int limit) {
+
+        if (limit > IdPManagementConstants.MAXIMUM_LIMIT_PER_PAGE) {
+            try {
+                String itemsPerPagePropertyValue = ServerConfiguration.getInstance()
+                        .getFirstProperty(IdPManagementConstants.ITEMS_PER_PAGE_PROPERTY);
```

Comment:
This is not the server max limit. This is the default server page limit.

Reply:
This is done

Reply:
No. It is still taking the default page limit when the limit is greater than Max allowed limit